Add collection conditions (#1387)
spockframework/spock: docs/utilities.adoc

There are many more ways to modify `MutableClock` just have a look at the JavaDocs, or the test code `spock.util.time.MutableClockSpec`.

[[collection-conditions]]
== Collection Conditions

> Sometimes, you want to assert the contents of a collection regardless of its order.

Author: Both sound weird to me, WDYT about
```suggestion
Sometimes, you want to assert the elements of a collection regardless of their order.
```

> Sometimes, you want to assert the contents of a collection regardless of its order.

Reviewer: SGTM

The Groovy way to do this is to cast both to `Set`, i.e. `x as Set == [1, 2, 3] as Set`.
While this works, it is very noisy.

Since Spock 2.1 you can use two new conditions:

> * `x =~ [1, 2, 3]` is a lenient match, i.e., casting both to `Set` before comparing.

Reviewer: ```suggestion
* `x =~ [1, 2, 3]` is a lenient match, i.e., checking that x contains at least one instance of every item in the list (same semantics as casting both to `Set` before comparing).
```

> * `x ==~ [1, 2, 3, 3]` is a strict match, using Hamcrest's `containsInAnyOrder` under the hood so that duplicate elements are verified as well.

Reviewer: ```suggestion
* `x ==~ [1, 2, 3, 3]` is a strict match, i.e., checking that x contains exactly the items in the list regardless of their order (using Hamcrest's `containsInAnyOrder` under the hood).
```
